api/src/server-utils.js

```diff
@@ -6,13 +6,17 @@ const logger = require('./logger');
 const MEDIC_BASIC_AUTH = 'Basic realm="Medic Web Services"';
 const cookie = require('./services/cookie');
 
-const wantsJSON = req => req.get('Accept') === 'application/json';
+const APP_JSON = 'application/json';
+
+const wantsJSON = req => req.get('Accept') === APP_JSON;
```

Comment:
It'd be slightly cleaner to use [`req.accepts`](https://expressjs.com/en/api.html#req.accepts) and [`res.type`](https://expressjs.com/en/api.html#res.type), or even [`res.format`](https://expressjs.com/en/api.html#res.format) if you're feeling ambitious!

Reply:
There's one issue with `req.accepts`. In the absence of an `Accepts` header, `req.accepts` will always return truthy, as anything passes as accepted, which is a bit different from what we're doing, where we specifically require the Accepts header to include 'application/json'. 

This changes behavior a little, including one e2e test that test for responses when json headers are not sent. 

Please let me know if you want to relax the accepts behavior for returning errors to requests.

Reply:
Let's not change the default behaviour, just in case...

res.format solves this by calling the first option. I haven't tested this but it might work..

```js
const writeHtml = (res, code, message, details) => {
  if (!res.headersSent) {
    res.writeHead(code, {
      'Content-Type': 'text/plain',
    });
  }
  if (message.message) {
    message = message.message;
  }
  if (typeof message !== 'string') {
    message = JSON.stringify(message);
  }
  if (details) {
    message += ': ' + JSON.stringify(details);
  }
  res.end(message);
};

const respond = (req, res, code, message, details) => {
  res.format({
    'text/html': () => {
      writeHtml(res, code, message, details);
    },
    'application/json': () => {
      writeJSON(res, code, message, details);
    }
  });
};
```

I'll leave it up to you whether that's better or not...

Reply:
From the documentation https://expressjs.com/en/api.html#res.format : 
```
The Content-Type response header is set when a callback is selected. However, you may alter this within the callback using methods such as res.set() or res.type().
```

Reply:
D'oh! Ok. What you had is perfect...